haystack/components/preprocessors/recursive_splitter.py

```diff
@@ -0,0 +1,408 @@
+    >Document(id=..., content: 'AI, in its broadest sense, is intelligence exhibited by machines, particularly computer systems.\n', meta: {'original_id': '65167a9823dd883de577e828ca4fd529e6f7241f0ff616acfce454d808478951', 'split_id': 1, 'split_idx_start': 45, '_split_overlap': []})
+    >Document(id=..., content: 'AI technology is widely used throughout industry, government, and science.', meta: {'original_id': '65167a9823dd883de577e828ca4fd529e6f7241f0ff616acfce454d808478951', 'split_id': 2, 'split_idx_start': 142, '_split_overlap': []})
+    >Document(id=..., content: ' Some high-profile applications include advanced web search engines; recommendation systems; interac...', meta: {'original_id': '65167a9823dd883de577e828ca4fd529e6f7241f0ff616acfce454d808478951', 'split_id': 3, 'split_idx_start': 216, '_split_overlap': []})
+    >]
+    """  # noqa: E501
+
+    def __init__(
+        self,
+        *,
+        split_length: int = 200,
+        split_overlap: int = 0,
+        split_unit: Literal["word", "char"] = "word",
+        separators: Optional[List[str]] = None,
+        sentence_splitter_params: Optional[Dict[str, Any]] = None,
+    ):
+        """
+        Initializes a RecursiveDocumentSplitter.
+
+        :param split_length: The maximum length of each chunk by default in characters, but can be in words.
+            See the `split_units` parameter.
+        :param split_overlap: The number of characters to overlap between consecutive chunks.
+        :param split_unit: The unit of the split_length parameter. It can be either "word" or "char".
+        :param separators: An optional list of separator strings to use for splitting the text. The string
+            separators will be treated as regular expressions unless the separator is "sentence", in that case the
+            text will be split into sentences using a custom sentence tokenizer based on NLTK.
+            See: haystack.components.preprocessors.sentence_tokenizer.SentenceSplitter.
+            If no separators are provided, the default separators ["\n\n", "sentence", "\n", " "] are used.
+        :param sentence_splitter_params: Optional parameters to pass to the sentence tokenizer.
+            See: haystack.components.preprocessors.sentence_tokenizer.SentenceSplitter for more information.
+
+        :raises ValueError: If the overlap is greater than or equal to the chunk size or if the overlap is negative, or
+                            if any separator is not a string.
+        """
+        self.split_length = split_length
+        self.split_overlap = split_overlap
+        self.split_units = split_unit
+        self.separators = separators if separators else ["\n\n", "sentence", "\n", " "]  # default separators
+        self.sentence_tokenizer_params = sentence_splitter_params
+        self._check_params()
+        self.nltk_tokenizer = None
+        if "sentence" in self.separators:
+            self.warm_up(sentence_splitter_params)
```

Comment:
I'd refactor this section. The goal of `warm_up` is to not have it called in the init method, but only by the Pipeline class. 

Here I'd set the `sentence_splitter_params` and enforce that keep_white_spaces is always True otherwise this component won't work properly right?

```suggestion
        self.sentence_splitter_params = sentence_splitter_params
        if self.sentence_splitter_params is None:
            self.sentence_splitter_params = {"keep_white_spaces": True}
        else:
            self.sentence_splitter_params["keep_white_spaces"] = True
```

Reply:
you are right, it should be the triggered externally 👍🏽 - thanks

Reply:
and yes, `keep_white_spaces` needs to be `True`, and It should be forced to True when `sentence` is a separator regardless of the value of the `sentence_splitter_params` values

Reply:
Hmm interesting, and when `keep_white_spaces` is set to `False` the splitting works as expected?

Reply:
or, better only if the `sentence_splitter_params` is not explicitly defined, we pass it, e.g.:

```python
self.sentence_splitter_params = {'keep_white_spaces: True'} if self.sentence_splitter_params is None else sentence_splitter_params
```

Reply:
Okay sounds good let's do as you suggest.

Reply:
> Hmm interesting, and when keep_white_spaces is set to False the splitting works as expected?

for this particularly case yes - I suspect it doesn't handle correctly something in that input test